pypesto/storage/read_from_hdf5.py

```diff
@@ -0,0 +1,89 @@
+import h5py
+from ..result import Result
+from ..optimize.result import OptimizerResult
+from ..problem import Problem
+from ..objective import Objective
+
+
+def read_hdf5_optimization(f: h5py.File,
+                           start: h5py.Group) -> 'OptimizerResult':
+    """
+    Read HDF5 results per start.
+
+    Parameters
+    ----------
+    f:
+        The HDF5 result file
+    start:
+        Specifies the start that is read from the HDF5 file
+    """
+
+    result = OptimizerResult()
+
+    for optimization_key in result.keys():
+        if optimization_key in f[f'/optimization/results/{start}']:
+            result[optimization_key] = \
+                f[f'/optimization/results/{start}/{optimization_key}'][:]
+        elif optimization_key in \
+                f[f'/optimization/results/{start}'].attrs:
+            result[optimization_key] = \
+                f[f'/optimization/results/{start}'].attrs[optimization_key]
+    return result
+
+
+class ProblemHDF5Reader:
+    """
+    Reader of the HDF5 problem files written
+    by class ProblemHDF5Writer.
+
+    Attributes
+    ---------
+    storage_filename:
+        HDF5 result file name
+    """
+    def __init__(self, storage_filename: str):
+        self.storage_filename = storage_filename
+        self.problem = Problem(Objective(), [], [])
+
+    def read(self) -> 'Problem':
```

Comment:
from ..problem import Problem
```suggestion
    def read(self) -> Problem:
```